components/forms/Register.tsx

```diff
@@ -0,0 +1,229 @@
+import { Dispatch, SetStateAction } from 'react';
+import styles from '@/styles/Login.module.css';
+import { createUserSchema, CreateUserInput } from '@/types/client/user';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { useSupabaseClient } from '@supabase/auth-helpers-react';
+import { Database } from '@/types/database.supabase';
+
+export default function Register({
+  setServerError,
+  setAuthType,
+}: {
+  setServerError: Dispatch<SetStateAction<string | null>>;
+  setAuthType: Dispatch<SetStateAction<'login' | 'register'>>;
+}) {
+  const supabase = useSupabaseClient<Database>();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<CreateUserInput>({
+    resolver: zodResolver(createUserSchema),
+  });
+
+  const onSubmit = async (formData: CreateUserInput) => {
+    const { data, error } = await supabase.auth.signUp({
+      email: formData.email,
+      password: formData.password,
+      options: {
+        data: {
+          username: formData.username,
+        },
+      },
+    });
+    if (error) {
+      setServerError(error.message);
```

Comment:
As mentioned in Discord, the form mentions incorrect errors here when either:
a) An email is already in use
b) A username is already in use

Where instead of the expected behaviour of mentioning the user/email/both are already in use, it reports the password must be longer than 3 characters (Speaking of, can we add a `TODO:` on that to change to a stronger value in future for prod?)